idea-plugin/src/main/kotlin/arrow/meta/ide/dsl/editor/color/ColorSettingsSyntax.kt

```diff
@@ -0,0 +1,277 @@
+   *     language = KotlinLanguage.INSTANCE,
+   *     highlighter = KotlinHighlighter(),
+   *     demoText = """
+   *       interface <$Interface>Functor</$Interface><F>
+   *       <$Keyword>extension</$Keyword> val semiSeven = 7.semigroup()
+   *       fun <F> hello(ctx: Env<F> = <$Keyword>given</$Keyword>): Kind<F, Unit> =
+   *       putStrLn(<$NamedArgument>str</$NamedArgument> = "hello")
+   *       <$Keyword>remote</$Keyword> fun distributed(): IO<Response>
+   *      """.trimIndent(),,
+   *     additionalHighlightingTags = mutableMapOf(
+   *      NamedArgument to KotlinHighlightingColors.NAMED_ARGUMENT,
+   *      Interface to DefaultLanguageHighlighterColors.INTERFACE_NAME,
+   *      Keyword to DefaultLanguageHighlighterColors.KEYWORD
+   *     )
+   *    )
+   *   )
+   *  }
+   *  //sampleEnd
+   * ```
+   *
+   * We can achieve a similar visual representation with an empty instance [PlainSyntaxHighlighter] - which is the default for [highlighter].
+   *
+   * ```kotlin:ank
+   * import arrow.meta.Plugin
+   * import arrow.meta.ide.IdeMetaPlugin
+   * import arrow.meta.invoke
+   * import com.intellij.psi.codeStyle.DisplayPriority
+   * import org.jetbrains.kotlin.idea.highlighter.KotlinHighlightingColors
+   *
+   * val Keyword: String = "keyword"
+   * val Interface: String = "Interface"
+   * val NamedArgument: String = "Named argument"
+   * val Number: String = "Number"
+   * val String: String = "String"
+   *
+   * //sampleStart
+   * val IdeMetaPlugin.nothingIdePlugin: Plugin
+   *  get() = "ColorSettingsPage" {
+   *   meta(
+   *    addColorSettingsPage(
+   *     displayName = "PlainColorSettings",
+   *     priority = DisplayPriority.LANGUAGE_SETTINGS,
+   *     demoText = """
+   *      <$Keyword>interface</$Keyword> <$Interface>Mappable</$Interface><F>
+   *      <$Keyword>extension</$Keyword> <$Keyword>val</$Keyword> semiSeven = <$Number>7</$Number>.semigroup()
+   *      <$Keyword>fun</$Keyword> hello(str: String): Unit = println(str)
+   *      <$Keyword>extension</$Keyword> <$Keyword>val</$Keyword> help: Unit = hello(<$NamedArgument>str</$NamedArgument> = <$String>"hello"</$String>)
+   *      <$Keyword>remote</$Keyword> <$Keyword>fun</$Keyword> distributed(): IO<Response>
+   *     """.trimIndent(),
+   *     additionalHighlightingTags = mutableMapOf(
+   *      NamedArgument to KotlinHighlightingColors.NAMED_ARGUMENT,
+   *      Interface to DefaultLanguageHighlighterColors.INTERFACE_NAME,
+   *      Keyword to DefaultLanguageHighlighterColors.KEYWORD,
+   *      Number to DefaultLanguageHighlighterColors.NUMBER,
+   *      String to DefaultLanguageHighlighterColors.STRING
+   *     )
+   *    )
+   *   )
+   *  }
+   *  //sampleEnd
+   * ```
+   * Ideally, constructing a [ColorSettingsPage] from a [SyntaxHighlighter] should be fairly linear as the latter already defines a Mapping between `Tokens` and `TextAttributes` in [SyntaxHighlighter.getTokenHighlights].
```

Comment:
@jansorg I wanted to add
```suggestion
   * Unsurprisingly, changes by the ide settings of programmatical or manual nature from the user won't affect the ide, because of the empty SyntaxHighlighter instance and there is no language specified for this [ColorSettingsPage]
   * Ideally, constructing a [ColorSettingsPage] from a [SyntaxHighlighter] should be fairly linear as the latter already defines a Mapping between `Tokens` and `TextAttributes` in [SyntaxHighlighter.getTokenHighlights].
```
Do you think that makes sens? I didn't have the time to validate that sentence throughout the whole internals, but I found hints which validate my assumption.